### PR TITLE
feat: update to work with new parser

### PR DESCRIPTION
**Pull request by obycode**
This corresponds to changes in hirosystems/clarity-repl#85.

### CODE REVIEW

**Comment by obycode** [comment]
These tests are expected to fail until the corresponding changes are merged into the REPL (hirosystems/clarity-repl#85).

File: src/types/project_manifest.rs
Comment (lgalabru) [inline src/types/project_manifest.rs:24]: What's your take on spawning a new section in the schema of the manifest, dedicated to tweaking the VM / REPL internals?

Comment (obycode) [reply src/types/project_manifest.rs:24]: Yes, that's probably a good idea. My check-checker changes (PR coming soon) is also tweaking this structure, to pass options directly to analysis passes:
```toml
[project.analysis_settings.check_checker]
trusted_sender = true
```

Do you think something like this would make sense:
```toml
[project]
name = "my-contract"
authors = []
description = ""
telemetry = false
requirements = []

[contracts.nft-trait]
path = "contracts/nft-trait.clar"
depends_on = []

[contracts.test]
path = "contracts/test.clar"
depends_on = ["nft-trait"]

[repl]
parser_version = 1
analysis = ["check_checker"]

[repl.analysis_settings.check_checker]
trusted_sender = true
```

**Comment by obycode** [comment]
Note - we also need to push a release of clarity-repl and update this Cargo.toml file accordingly before pushing this.

File: src/types/project_manifest.rs
Comment (obycode) [reply src/types/project_manifest.rs:24]: Now I'm thinking the `analysis_settings` struct is not necessary and it could just be `repl.check_checker`.

Comment (obycode) [reply src/types/project_manifest.rs:24]: Ok, actually I have a much nicer fix to this in my check-checker branch. Mind if we hold off on this change until that one?

**Review by lgalabru — APPROVED** [review]
This PR looks good to me, thank you @obycode!
Before merging, could you do a sanity pass and make sure that the LSP is still functional with these changes?
Thanks!

File: src/types/project_manifest.rs
Comment (obycode) [reply src/types/project_manifest.rs]: This is implemented in the latest commits.

**Review by lgalabru — DISMISSED** [review]

File: src/generate/project.rs
Comment (lgalabru) [inline src/generate/project.rs:178]: What's your take on flattening a bit the schema here? `repl.analysis`, instead of `project.repl.analysis`?

Comment (obycode) [reply src/generate/project.rs:178]: Sure, that makes sense. I will quickly make that change and push.

Comment (obycode) [reply src/generate/project.rs]: Changed in a569502

**Review by lgalabru — APPROVED** [review]
Looks great, thank you @obycode!

**Comment by unclemantis** [comment]
We stable yet?

**Comment by obycode** [comment]
> We stable yet?
> […](#)

😄 Yup, merging and releasing now!

**Pull request merged**
